pkg/aws/metadata/handler_credentials.go

```diff
@@ -50,34 +50,10 @@ func (c *credentialsHandler) Handle(ctx context.Context, w http.ResponseWriter,
 		return http.StatusInternalServerError, err
 	}
 
-	foundRole, err := findRole(ctx, c.roleFinder, ip)
-	if err != nil {
-		metrics.GetOrRegisterMeter("credentialsHandler.findRoleError", metrics.DefaultRegistry).Mark(1)
-		return http.StatusInternalServerError, fmt.Errorf("error finding pod for ip %s: %s", ip, err.Error())
-	}
-
-	if foundRole == "" {
-		metrics.GetOrRegisterMeter("credentialsHandler.emptyRole", metrics.DefaultRegistry).Mark(1)
-		return http.StatusNotFound, EmptyRoleError
-	}
-
 	requestedRole := mux.Vars(req)["role"]
-	if requestedRole == "" {
-		return http.StatusBadRequest, fmt.Errorf("no role specified")
-	}
-
-	decision, err := c.policy.IsAllowedAssumeRole(ctx, requestedRole, ip)
-	if err != nil {
-		return http.StatusInternalServerError, fmt.Errorf("error checking assume role permissions: %s", err.Error())
-	}
-
-	if !decision.IsAllowed() {
-		return http.StatusForbidden, fmt.Errorf("assume role forbidden: %s", decision.Explanation())
-	}
-
-	credentials, err := c.credentialsForRole(ctx, requestedRole)
+	credentials, err := c.fetchCredentials(ctx, ip, requestedRole)
```

Comment:
There's no more code that can be deleted from this handler without breaking it. Good stuff.